features/netsocket/Socket.h

```diff
@@ -163,8 +163,32 @@ class Socket {
      *  The callback may be called in an interrupt context and should not
      *  perform expensive operations such as recv/send calls.
      *
+     *  Note! This is not intended as a replacement for a poll or attach-like
+     *  asynchronous api, but rather as a building block for constructing
+     *  such functionality. The exact timing of when the registered function
+     *  is called is not garunteed and susceptible to change.
+     *
+     *  @param func     Function to call on state change
+     */
+    void sigio(mbed::Callback<void()> func);
+
+    /** Register a callback on state change of the socket
+     *
+     *  The specified callback will be called on state changes such as when
+     *  the socket can recv/send/accept successfully and on when an error
+     *  occurs. The callback may also be called spuriously without reason.
+     *
+     *  The callback may be called in an interrupt context and should not
+     *  perform expensive operations such as recv/send calls.
+     *
      *  @param func     Function to call on state change
+     *  @deprecated
```

Comment:
Seems misleading to have a whole duplicate comment block (to me at least). Implies it somehow might have slightly different functionality from sigio, and we need to compare the comments to spot the difference.

Can't you just have some sort of reference link?